packages/TimePicker/src/TimePicker.js

```diff
@@ -137,94 +147,71 @@ export default class TimeInput extends React.Component {
       return;
     }
 
-    this.setState({
-      isTabIndexActive: 0,
-      isVisible: false,
-    });
+    setIsTabIndexActive(0);
+    setIsOpen(false);
   };
 
-  handleClickOutside = () => {
-    this.finish();
-  };
+  function finish() {
+    setIsOpen(false);
+    setValue(timeStr);
+  }
 
-  handleKeyUp = event => {
+  const handleKeyUp = event => {
     if (event.key === "Enter") {
-      this.finish();
+      finish();
     }
 
     if (event.key === "Escape") {
-      this.finish();
+      finish();
     }
   };
 
-  handleError(time) {
-    this.setState({
-      ...time,
-      error: this.props.labelError,
-      isVisible: false,
-    });
-  }
+  const { t } = useI18n();
+  return (
+    <sc.TimePicker onFocus={handleFocus} onBlur={handleBlur}>
+      <Popover style={{ width: "100%" }} isOpen={isOpen} edge="left" offset={0} align="bottom">
+        {/* eslint-disable-next-line */}
+        <div tabIndex={isTabIndexActive ? 0 : -1}>
+          {/* setting hasClearButton to false confuses, look like a close button for the timeinput hasClearButton */}
+          <Popover.Trigger style={{ width: "100%" }}>
+            <Input
+              ariaLabel={a11yText}
+              hasClearButton={false}
+              icon={<SvgClockTime />}
+              isDisabled={isDisabled}
+              onChange={handleChange}
```

Comment:
Are you finding that the `onChange` is working as you would expect and don't need to use `onInput` now? (https://github.com/acl-services/paprika/pull/639/files#diff-f248e54c9705d57a299d0e5b2a13b00bL200)

Reply:
I changed it, I've never used `onInput` before, and the `onChange` here doesn't cause any issues.

Reply:
Ok, I think @nahumzs was having some issues with it, I was going to look into what that issue was but perhaps it's not needed

Reply:
yes, onChanged was having issues but maybe was also the way was previously structured coded, but if its working now perfect! 👏